pkg/internalmetrics/metrics.go

```diff
@@ -0,0 +1,55 @@
+package internalmetrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	ControlPlaneModeValueClusterScoped = "ClusterWide"
+	ControlPlaneModeValueMultiTenant   = "MultiTenant"
+)
+
+// These constants define the different label names for the different metric timeseries
+const (
+	labelSMCPNamespace = "smcp_namespace"
+	labelSMCPVersion   = "smcp_version"
+	labelSMCPMode      = "smcp_mode"
+)
+
+// MetricsType defines all of istio-operator's metrics.
+type MetricsType struct {
+	MemberCounter *prometheus.GaugeVec
+}
+
+// Metrics contains all of istio-operator's own internal metrics.
+// These metrics can be accessed directly to update their values, or
+// you can use available utility functions defined below.
+var Metrics = MetricsType{
+	MemberCounter: prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "servicemesh_member_count",
```

Comment:
you don't want to name it with a `_count` suffix if it's not a counter. `servicemesh_members` would be a better choice.

Reply:
or even `servicemesh_namespaces`.

Reply:
Actually it is the number of members in the mesh/meshes. I didn't use a `prometheus.Counter` type because I needed to reset it in specific cases but it is used like a counter